Carbons with NO_HUNGER can regenerate blood

KittyNoodle opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
Makes Carbons with NO_HUNGER regenerate blood

Before you get mad cobby the issue that this is fixing was set as a bug.
## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Permanent blood loss is bad. 
Fixes https://github.com/tgstation/tgstation/issues/62492

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. -->

:cl:
fix: Non eaters rejoice as you can now actually regenerate blood.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

KittyNoodle committed:
Fixes mobs with NO_HUNGER not regenerating blood

KittyNoodle committed:
Merge branch 'tgstation:master' into Bloodregenfix

KittyNoodle committed:
Merge branch 'tgstation:master' into Bloodregenfix

KittyNoodle commented on code/modules/mob/living/blood.dm:
```diff
@@ -30,6 +30,9 @@
 			adjust_nutrition(-nutrition_ratio * HUNGER_FACTOR * delta_time)
 			blood_volume = min(blood_volume + (BLOOD_REGEN_FACTOR * nutrition_ratio * delta_time), BLOOD_VOLUME_NORMAL)
+		
+		if(blood_volume < BLOOD_VOLUME_NORMAL && HAS_TRAIT(src, TRAIT_NOHUNGER)) //blood regen for non eaters
```
Vamps are such a niche species, just gonna add the amount of blood regenerated to the amount removed for vamps, they don’t need a whole trait for one species.

KittyNoodle replied:
Done

KittyNoodle committed:
It’s snowflake code, but everything else about vampire is snowflake so i guess this fits

Ghommie commented on code/modules/mob/living/carbon/human/species_types/vampire.dm:
```diff
@@ -51,6 +51,7 @@
 		C.adjustCloneLoss(-2 * delta_time)
 		return
 	C.blood_volume -= 0.125 * delta_time
+	C.blood_volume -= BLOOD_REGEN_FACTOR * 1.25 * delta_time //snowflake code so vamps do not regen blood.
```
```suggestion
	C.blood_volume -= BLOOD_REGEN_FACTOR  * 1.375 * delta_time //snowflake code so vamps do not regen blood.
```
😕

KittyNoodle replied:
Yea

KittyNoodle committed:
Update code/modules/mob/living/carbon/human/species_types/vampire.dm

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

KittyNoodle commented:
TFW i forget basic arithmetic

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/vampire.dm:
```diff
@@ -50,7 +50,7 @@
 		C.adjustCloneLoss(-2 * delta_time)
 		return
-	C.blood_volume -= 0.125 * delta_time
+	C.blood_volume -= BLOOD_REGEN_FACTOR  * 1.375 * delta_time //snowflake code so vamps do not regen blood.
```
but now vampires who've consumed ghoul powder, are cold, are in a regenerative coma, etc. will lose blood faster

more importantly, vampires who're above BLOOD_VOLUME_NORMAL (which you'll almost always be after you drain a monkey or two) will lose blood at a faster rate than normal

also, have you checked to see if this will cause your blood level thing in the status menu to fluctuate up and down?

KittyNoodle replied:
> also, have you checked to see if this will cause your blood level thing in the status menu to fluctuate up and down?

will test that later, they _should_ be on the same tick.

I really don’t wanna make a trait that only one janky race uses.

KittyNoodle committed:
hacky fix

KittyNoodle commented:
Fix too jank

KittyNoodle commented:
Must fix fix

KittyNoodle committed:
Fixes hacky fix

Apparently for some ungodly reason coretemperature is defined as bodytemperature in some places

KittyNoodle committed:
The final fix(i was just being stupid)

KittyNoodle commented:
Coders plz

optimumtact commented:
15 days to fix a merge conflict.
